Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+					ShowScenesBox();
+					g_GUISound.BackButtonPressSound()->Play();
+				} else if (eventControl == m_ScenarioButtons.at(ScenarioButtons::StartGameButton)) {
+					if (StartGame()) {
+						HideAllScreens();
+						m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->SetVisible(true);
+						g_GUISound.ButtonPressSound()->Play();
+						return ScenarioUpdateResult::ActivityRestarted;
+					} else {
+						g_GUISound.UserErrorSound()->Play();
 					}
 				}
 			}
+			if (dynamic_cast<const GUIButton *>(eventControl) && guiEvent.GetType() == GUIEvent::Notification && guiEvent.GetMsg() == GUIButton::Focused) { g_GUISound.SelectionChangeSound()->Play(); }
+		}
+		return ScenarioUpdateResult::NoEvent;
+	}
 
-			// Move the dot closer to the planet center
-			Vector offsetIncrement;
-			if ((*pItr)->GetLocation().GetY() > 0)
-				offsetIncrement = Vector(0, -8);
-			else
-				offsetIncrement = Vector(0, 8);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-			if (isOverlapped)
-			{
-				if (abs((*pItr)->GetLocation().GetY()) > m_PlanetRadius)
-					offsetIncrement.m_Y = -offsetIncrement.m_Y * 2;
+	void ScenarioGUI::Update() {
+		m_Controller->Update();
 
-				(*pItr)->SetLocationOffset((*pItr)->GetLocationOffset() + offsetIncrement);
+		if (g_ConsoleMan.IsEnabled() && !g_ConsoleMan.IsReadOnly()) {
+			return;
+		}
 
+		if (m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->GetVisible()) {
+			if (m_ScenarioButtons.at(ScenarioButtons::ResumeButton)->GetVisible()) { m_ScenarioGUIController->GetManager()->SetFocus((m_BlinkTimer.AlternateReal(500)) ? m_ScenarioButtons.at(ScenarioButtons::ResumeButton) : nullptr); }
+			if (m_ScenarioCollectionBoxes.at(ScenarioCollections::SceneInfoBox)->GetVisible()) { m_ScenarioButtons.at(ScenarioButtons::StartHereButton)->SetText(m_BlinkTimer.AlternateReal(333) ? "Start Here" : "> Start Here <"); }
+		} else if (m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox)->GetVisible()) {
+			UpdatePlayersBox();
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::DrawSitePoints(BITMAP *drawBitmap) const {
+		if (m_ScenarioScenes && m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->GetVisible()) {
+			for (const Scene *scenePointer : *m_ScenarioScenes) {
+				// Mark user-created scenes to let players distinguish them from built-in.
+				int drawColor = (scenePointer->GetModuleID() == g_PresetMan.GetModuleID("Scenes.rte")) ? c_GUIColorRed : c_GUIColorYellow;
+
+				Vector sitePos(m_PlanetCenter + scenePointer->GetLocation() + scenePointer->GetLocationOffset());
+				int sitePosX = sitePos.GetFloorIntX();
+				int sitePosY = sitePos.GetFloorIntY();
+
+				int blendAmount = 70 + RandomNum(0, 40);
```

Comment:
Huh, I'd never noticed that scenes had randomized blending/brightness. Seems a little pointless to me lol, given how unnoticeable it is. Or is that just a me thing?

Reply:
It's a you thing and it's not going away.